src/LoginForm.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useContext, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { createUser, getUser } from './api';
+import { FormInput } from './FormInput';
+import { Loader } from './Loader';
+import './styles/LoginForm.scss';
+import { TodosContext } from './TodosProvider';
+
+export const LoginForm = React.memo(() => {
```

Comment:
looks like you can split this component into smaller - it will be easier to read your code

Reply:
and create separate methods for handling logic and then use them in jsx